pkg/operator/tpr.go

```diff
@@ -101,18 +110,36 @@ func waitForTPRInit(context *context, tpr TPR) error {
 	})
 }
 
-func watchTPR(context *context, name string, resourceVersion string) (*http.Response, error) {
-	return context.kubeHttpCli.Get(fmt.Sprintf("%s/%s?watch=true&resourceVersion=%s",
-		context.masterHost, tprURI(context, name), resourceVersion))
+func watchTPRNamespaced(context *context, name, namespace, resourceVersion string) (*http.Response, error) {
+	uri := fmt.Sprintf("%s/%s?watch=true&resourceVersion=%s", context.masterHost, tprURINamespaced(name, namespace), resourceVersion)
+	logger.Infof("watching tpr: %s", uri)
+	return context.kubeHttpCli.Get(uri)
 }
 
-func tprURI(context *context, name string) string {
-	return fmt.Sprintf("/apis/%s/%s/namespaces/%s/%ss", tprGroup, tprVersion, context.namespace, name)
+func watchTPR(context *context, name, resourceVersion string) (*http.Response, error) {
+	uri := fmt.Sprintf("%s/%s?watch=true&resourceVersion=%s", context.masterHost, tprURI(name), resourceVersion)
+	logger.Debugf("watching tpr: %s", uri)
+	return context.kubeHttpCli.Get(uri)
 }
 
-func getRawList(context *context, tpr TPR) ([]byte, error) {
-	restcli := context.clientset.CoreV1().RESTClient()
-	return restcli.Get().RequestURI(tprURI(context, tpr.Name())).DoRaw()
+func tprURINamespaced(name, namespace string) string {
+	return fmt.Sprintf("/apis/%s/%s/namespaces/%s/%ss", tprGroup, tprVersion, namespace, name)
```

Comment:
not a typo. The TPRs are actually made plural for a RESTful api. I was very confused too. :)